lib/liquid/variable_lookup.rb

```diff
@@ -54,6 +54,8 @@ def evaluate(context)
         elsif @command_flags & (1 << i) != 0 && object.respond_to?(key)
           object = object.send(key).to_liquid
 
+        elsif context.strict_variables
+          raise Liquid::UndefinedVariable, "undefined variable #{key} in #{object}"
```

Comment:
I'm still worried about including `#{object}` in the string. I don't see the benefit, but I do see potential leaks.